src/adapter/src/coord/peek.rs

```diff
@@ -88,8 +88,26 @@ impl FastPathPlan {
                         .unwrap_or_else(|| index.to_string())
                 )
                 .unwrap();
-                if let Some(value) = value {
-                    writeln!(&mut explanation, "| | Lookup value {}", value).unwrap();
+                if let Some(literal_constraints) = literal_constraints {
+                    write!(&mut explanation, "| | Lookup ").unwrap();
+                    if literal_constraints.len() == 1 {
+                        writeln!(
+                            &mut explanation,
+                            "value {}",
+                            literal_constraints.get(0).unwrap()
+                        )
+                        .unwrap();
+                    } else {
+                        assert!(literal_constraints.len() > 1);
+                        write!(&mut explanation, "values [").unwrap();
+                        for (i, l) in literal_constraints.iter().enumerate() {
+                            write!(&mut explanation, "{}", l).unwrap();
+                            if i < literal_constraints.len() - 1 {
+                                write!(&mut explanation, "; ").unwrap();
+                            }
+                        }
+                        writeln!(&mut explanation, "]").unwrap();
```

Comment:
As commented in the other PR, it would be nice to use `mz_ore::str::{separated, bracketed}`

Reply:
Done